enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/mtls/MTLSAuthenticator.java

```diff
@@ -0,0 +1,190 @@
+import org.wso2.choreo.connect.enforcer.constants.APISecurityConstants;
+import org.wso2.choreo.connect.enforcer.security.Authenticator;
+import org.wso2.choreo.connect.enforcer.tracing.TracingConstants;
+import org.wso2.choreo.connect.enforcer.tracing.TracingSpan;
+import org.wso2.choreo.connect.enforcer.tracing.TracingTracer;
+import org.wso2.choreo.connect.enforcer.tracing.Utils;
+import org.wso2.choreo.connect.enforcer.util.FilterUtils;
+
+import java.security.KeyStore;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Map;
+
+/**
+ * Implements the authenticator interface to authenticate request using MTLS.
+ */
+public class MTLSAuthenticator implements Authenticator {
+    private static final Logger log = LogManager.getLogger(MTLSAuthenticator.class);
+    private final boolean isEnableClientValidation = FilterUtils.getClientValidationStatus();
+    private final boolean isClientCertificateEncode = FilterUtils.getCertificateEncodeStatus();
+
+    @Override
+    public boolean canAuthenticate(RequestContext requestContext) {
+        String cert = "";
+
+        if (isEnableClientValidation) {
+            cert = requestContext.getClientCertificate();
+        } else {
+            if (requestContext.getHeaders().containsKey(FilterUtils.getCertificateHeaderName())) {
+                cert = retrieveMTLSHeaderValue(requestContext);
+            }
+        }
+        if (cert.isEmpty()) {
+            log.debug("Could +not find a valid client certificate in the request: " + requestContext.getRequestPath());
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public AuthenticationContext authenticate(RequestContext requestContext) throws APISecurityException {
+        TracingTracer tracer;
+        TracingSpan mtlsApiAuthenticatorSpan = null;
+        Scope mtlsApiAuthenticatorSpanScope = null;
+
+        try {
+            if (Utils.tracingEnabled()) {
+                tracer = Utils.getGlobalTracer();
+                mtlsApiAuthenticatorSpan = Utils.startSpan(TracingConstants.MTLS_API_AUTHENTICATOR_SPAN, tracer);
+                mtlsApiAuthenticatorSpanScope = mtlsApiAuthenticatorSpan.getSpan().makeCurrent();
+                Utils.setTag(mtlsApiAuthenticatorSpan, APIConstants.LOG_TRACE_ID,
+                        ThreadContext.get(APIConstants.LOG_TRACE_ID));
+            }
+
+            AuthenticationContext authenticationContext = new AuthenticationContext();
+            KeyStore trustStore = retrieveTrustStore(requestContext);
+            Map<String, String> mtlsCertificateTiers = retrieveMtlsCertificateTiers(requestContext);
+            X509Certificate clientCert;
+            boolean authenticated = false;
+            String clientCertificateTier;
+            String clientCertificateAlias;
+
+            try {
+                clientCert = getClientCertificate(requestContext);
```

Comment:
can make X509Certificate clientCert; a local variable here. can do same for clientCertificateTier, clientCertificateAlias
```suggestion
                X509Certificate clientCert = getClientCertificate(requestContext);
```

Reply:
Fixed